include/odp/api/spec/init.h

```diff
@@ -332,6 +332,16 @@ int odp_term_local(void);
  */
 int odp_term_global(odp_instance_t instance);
 
+/**
+ * Set log function
+ *
+ * Sets the log function for the calling thread. Setting the log function to
+ * NULL causes the calling thread to use the global log function.
```

Comment:
I think the spec text could be more explicit that the intention of this call is to override the global log function (and its usage is optional). May be something like this:

"Set per thread log function

By default, all ODP log writes use the global log function, which may be set as part of odp_init_t. Using this operation, an alternative ODP log function may be set for the calling thread. When set, ODP uses the thread specific log function for all log writes originating from ODP API calls made by the calling thread. Setting the log function to NULL ..."

Reply:
Fixed in v4 (with "Set thread specific log function").